### ISSUE TITLE
Android 9 crashes on camera

### ISSUE THREAD

**Issue by YoussifHamdy** [ArthurHub/Android-Image-Cropper#718]
After capturing an image using the camera and click OK then it's crash and throwing this exception ..
android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/${applicationId}/cache/pickImageResult.jpeg exposed beyond app through Intent.getData()

**Comment by Stinox22** [comment]
Check this article
https://stackoverflow.com/questions/38200282/android-os-fileuriexposedexception-file-storage-emulated-0-test-txt-exposed

**TLDR;
In your Manifest.xml**
`<provider
            android:name=".util.GenericFileProvider"
            android:authorities="${applicationId}.provider"
            android:exported="false"
            android:grantUriPermissions="true">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/providers_path"/>
        </provider>
`


**util/GenericFileProvider.java**
`import androidx.core.content.FileProvider;
public class GenericFileProvider extends FileProvider {}`

**res/xml/providers_path.xml**
`<?xml version="1.0" encoding="utf-8"?>
<paths xmlns:android="http://schemas.android.com/apk/res/android">
    <external-path name="external_files" path="."/>
</paths>`

**Comment by Canato** [comment]
please check [this](https://github.com/ArthurHub/Android-Image-Cropper/issues/776#issuecomment-728336763)
